cni/network/network.go

```diff
@@ -607,13 +639,50 @@ func (plugin *NetPlugin) createNetworkInternal(
 	nwInfo := network.NetworkInfo{}
 	ipamAddResult.hostSubnetPrefix.IP = ipamAddResult.hostSubnetPrefix.IP.Mask(ipamAddResult.hostSubnetPrefix.Mask)
 	ipamAddConfig.nwCfg.IPAM.Subnet = ipamAddResult.hostSubnetPrefix.String()
+
+	ipnet := net.IPNet{}
+	interfaceInfo := network.InterfaceInfo{}
+	// find ipnet from secondaryInterfaceInfo if it exists;
+	if hasSecondaryInterface(ipamAddResult) {
+		interfaceInfo = ipamAddResult.secondaryInterfacesInfo[0]
+		_, subnet, err := net.ParseCIDR(ipamAddResult.secondaryInterfacesInfo[0].IPConfigs[0].Address.String())
+		if err != nil {
+			logger.Error("Failed to parse secondary interface ipnet", zap.Error(err))
+		}
+		ipnet = *subnet
+	} else {
+		interfaceInfo = ipamAddResult.defaultInterfaceInfo
+		ipnet = ipamAddResult.hostSubnetPrefix
+	}
+
 	// Find the master interface.
-	masterIfName := plugin.findMasterInterface(ipamAddConfig.nwCfg, &ipamAddResult.hostSubnetPrefix)
+	var macAddress string
+	if ipamAddResult.secondaryInterfacesInfo != nil {
+		macAddress = ipamAddResult.secondaryInterfacesInfo[0].MacAddress.String()
+	}
+
+	masterIfName := plugin.findMasterInterface(ipamAddConfig.nwCfg, &ipnet)
 	if masterIfName == "" {
-		err := plugin.Errorf("Failed to find the master interface")
-		return nwInfo, err
+		// check if masterInterface is in SecondaryInterfaces map
+		// if not, then no master interface is found
+		if SecondaryInterfaces != nil {
+			for name, secondaryInterface := range SecondaryInterfaces {
+				if secondaryInterface.macAddress == macAddress {
+					masterIfName = name
+				}
+			}
+		} else {
+			err := plugin.Errorf("Failed to find the master interface")
+			return nwInfo, err
+		}
 	}
-	logger.Info("Found master interface", zap.String("ifname", masterIfName))
+
+	var secondaryInterfaceProperty SecondaryInterfaceProperty
+	secondaryInterfaceProperty.ipNet = ipnet.String()
+	secondaryInterfaceProperty.macAddress = macAddress
+
+	SecondaryInterfaces = make(map[string]SecondaryInterfaceProperty)
+	SecondaryInterfaces[masterIfName] = secondaryInterfaceProperty
```

Comment:
this code is not required

Reply:
just remembered this code should be here because when a pod is created, then we cannot see that interface from ipconfig; so we must have a map to record which interface is being used, otherwise we cannot find master interface by macAddress because ipconfig will return nil to us;